srcs/http/response/http_path_check/http_path_check.cpp

```diff
@@ -0,0 +1,107 @@
+#include "http_path_check.hpp"
+#include <cstdlib> // atoi
+#include <iostream>
+
+namespace http {
+
+CheckPathResult HttpPathCheck::Check(const MockDtoServerInfos &server_info, HttpRequest &request) {
+	CheckPathResult result;
+
+	CheckDTOServerInfo(result, server_info, request.header_fields);
+	CheckLocationList(result, server_info.locations, request.request_line.request_target);
+	return result;
+}
+
+// Check Server_info
+void HttpPathCheck::CheckDTOServerInfo(
+	CheckPathResult &result, const MockDtoServerInfos &server_info, HeaderFields &header_fields
+) {
+	if (server_info.host != header_fields["Host"]) { // Check host_name
+		result.is_ok = CheckPathResult::INVALID_HOST;
+	} else if (static_cast<size_t>(std::atoi(header_fields["Content-Length"].c_str())) >
+			   server_info.client_max_body_size) { // Check content_length
+		result.is_ok = CheckPathResult::PAYLOAD_TOO_LARGE;
+	} else if (!server_info.error_page.second.empty()) { // Check error_page
+		result.error_status_code = server_info.error_page.first;
+		result.error_page_path   = server_info.error_page.second;
+	}
+	return;
+}
+
+// Check LocationList
+void HttpPathCheck::CheckLocationList(
+	CheckPathResult &result, const LocationList &locations, const std::string &request_target
+) {
+	if (result.is_ok != CheckPathResult::OK) {
+		return;
+	}
+	MockLocationCon match_location = CheckLocation(result, locations, request_target);
+	std::cout << match_location.request_uri << std::endl; // for debug
+	CheckIndex(result, match_location);
+	CheckAutoIndex(result, match_location);
+	CheckAlias(result, match_location);
+	CheckRedirect(result, match_location);
+	return;
+}
+
+MockLocationCon HttpPathCheck::CheckLocation(
+	CheckPathResult &result, const LocationList &locations, const std::string &request_target
+) {
+	MockLocationCon match_loc;
+
+	// ex. request.target /www/target.html:
+	// uri = /www/, target = target.html
+	// location1 /www/data/
+	// location1 /www/
+	// location2 /
+	// -> /www/
+	result.path = request_target;
+	for (LocationList::const_iterator it = locations.begin(); it != locations.end(); ++it) {
+		if (request_target.find((*it).request_uri) == 0 &&
+			(*it).request_uri.length() > match_loc.request_uri.length()) { // Longest Match
+			match_loc = *it;
+		}
+	}
+	if (match_loc.request_uri.length() == 0) {
```

Comment:
本当にどっちでも良いのですが、他を empty() に修正してるみたいだったので、ここも一緒でも良さそう
```suggestion
	if (match_loc.request_uri.empty()) {
```